src/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -183,6 +183,14 @@ internal void SetRefreshCredentialNeeded()
             _refreshCredentialNeeded = true;
         }
 
+        ~SecureChannel()
+        {
+            if(_sslAuthenticationOptions.AlpnProtocolsHandle.IsAllocated)
```

Comment:
Nit: space after the `if`

Reply:
This has the same issue you were previously concerned about, that AlpnProtocolsHandle.Free doesn't mutate the actual field but rather the returned copy.  And in this case that seems like it could actually matter, e.g. if an exception is thrown in Close() after free'ing the handle but before SuppressFinalize is called, in which case the finalize will still run and this will attempt to free it again.

Reply:
I have tested it over and over, it actually can be double free'd the issue I had previously was reusing it AFTER I had freed it. So I think this is fine, with the idea that it should go very soon anyway, and we don't check this before we use it.